File: 1051-Height-Checker/1051-Height-Checker.java
class Solution {
    public int heightChecker(int[] heights) {
        int[] cache = new int[101];
        for(int i = 0; i < heights.length; i++){
            cache[heights[i]] += 1;
        }
        int[] sorted = new int[heights.length];
        int pointer = 0;
        for(int i = 0; i < 101; i++){
            while(cache[i] > 0){
                sorted[pointer] = i;
                pointer++;
                cache[i] -= 1;
            }
        }
        int count = 0;
        for(int i = 0; i < heights.length; i++){
            if(heights[i] != sorted[i]){
                count++;
            }
        }
        return count;
    }
}

File: 1051-Height-Checker/1051-Height-Checker.py
class Solution:
    def heightChecker(self, heights: List[int]) -> int:
        backup = heights.copy()
        heights.sort()
        count = 0
        print(backup)
        print(heights)
        for i in range(len(heights)):
            if heights[i] != backup[i]:
                count += 1
        return count